src/main/scala/br/unb/cic/soot/svfa/jimple/JSVFA.scala

```diff
@@ -171,38 +172,62 @@ abstract class JSVFA extends SVFA with Analysis with FieldSensitiveness with Obj
     }
   }
 
-
   def createSceneTransform(): (String, Transform) = ("wjtp", new Transform("wjtp.svfa", new Transformer()))
 
   def initAllocationSites(): Unit = {
     val listener = Scene.v().getReachableMethods.listener()
 
     while(listener.hasNext) {
       val m = listener.next().method()
-      if (m.hasActiveBody) {
-        val body = m.getActiveBody
-        body.getUnits.forEach(unit => {
-          if (unit.isInstanceOf[soot.jimple.AssignStmt]) {
-            val right = unit.asInstanceOf[soot.jimple.AssignStmt].getRightOp
-            if (right.isInstanceOf[NewExpr] || right.isInstanceOf[NewArrayExpr] || right.isInstanceOf[StringConstant]) {
-              allocationSites += (right -> createNode(m, unit))
-            }
+
+      updateAllocationSites(m)
+
+    }
+  }
+
+  def updateAllocationSites(m: SootMethod): Unit = {
+
+    if(visitedMethodsAllocationSites.contains(m)) {
+      return
+    }
+
+    visitedMethodsAllocationSites.add(m)
+
+    if (m.hasActiveBody) {
+      val body = m.getActiveBody
+      body.getUnits.forEach(unit => {
+        if (unit.isInstanceOf[soot.jimple.AssignStmt]) {
+          val right = unit.asInstanceOf[soot.jimple.AssignStmt].getRightOp
+          if (right.isInstanceOf[soot.jimple.InstanceInvokeExpr]){
+            val method = right.asInstanceOf[soot.jimple.InstanceInvokeExpr]
+            updateAllocationSites(method.getMethod)
           }
-          else if(unit.isInstanceOf[soot.jimple.ReturnStmt]) {
-            val exp = unit.asInstanceOf[soot.jimple.ReturnStmt].getOp
-            if(exp.isInstanceOf[StringConstant]) {
-              allocationSites += (exp -> createNode(m, unit))
-            }
+          if (right.isInstanceOf[NewExpr] || right.isInstanceOf[NewArrayExpr] || right.isInstanceOf[StringConstant]) {
+            allocationSites += (right -> createNode(m, unit))
           }
-        })
-      }
+        }
+        else if(unit.isInstanceOf[soot.jimple.ReturnStmt]) {
+          val exp = unit.asInstanceOf[soot.jimple.ReturnStmt].getOp
+          if(exp.isInstanceOf[StringConstant]) {
+            allocationSites += (exp -> createNode(m, unit))
+          }
+        }
+
+        if (unit.isInstanceOf[soot.jimple.InvokeStmt]){
```

Comment:
Se for apenas uma chamada direta de método, atualiza os allocation sites. Por exemplo: stack1.m()